File: backend/models/review.go
package models

import (
	"time"
)

type Review struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time `json:"created_at" time_format:"2006-01-02T15:04:05Z07:00"`
	UpdatedAt time.Time `json:"updated_at" time_format:"2006-01-02T15:04:05Z07:00"`
	UserID    uint      `json:"user_id" validate:"required"`
	User      User      `json:"user" gorm:"foreignKey:UserID"`
	HikeID    uint      `json:"hike_id" validate:"required"`
	Rating    int       `json:"rating" validate:"required,min=1,max=5"`
	Comment   string    `json:"comment" validate:"max=280"`
}

File: frontend/lib/shared/models/review.dart
import 'package:frontend/shared/models/user.dart';

class Review {
  final int id;
  final int userId;
  final int hikeId;
  final int rating;
  final String comment;
  final DateTime createdAt;
  final DateTime updatedAt;
  final User user;

  Review({
    required this.id,
    required this.userId,
    required this.hikeId,
    required this.rating,
    required this.comment,
    required this.createdAt,
    required this.updatedAt,
    required this.user,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      userId: json['user_id'],
      hikeId: json['hike_id'],
      rating: json['rating'],
      comment: json['comment'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
      user: User.fromJson(json['user']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'hike_id': hikeId,
      'rating': rating,
      'comment': comment,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'user': user.toJson(),
    };
  }
}
